curriculum/challenges/english/20-upcoming-python/learn-classes-and-objects-by-building-a-sudoku-solver/6569d83fe4dcc614c2ff971d.md

```diff
@@ -0,0 +1,81 @@
+---
+id: 6569d83fe4dcc614c2ff971d
+title: Step 41
+challengeType: 20
+dashedName: step-41
+---
+
+# --description--
+
+For each element in the specified column (`col`) of the current row (`row`), check whether the value at the current position in the 2D list is not equal to the provided `num`.
+
+# --hints--
+
+Access the element at the current position in the 2D list using `self.board[row][col]`.
+
+```js
+assert.match(code, /self\.board\s*\[\s*row\s*\]\s*\[\s*col\s*\]/);
+```
+
+Check if the element at the current position in the 2D list is not equal to the provided `num`.
+
+```js
+assert.match(code, /self\.board\s*\[\s*row\s*\]\s*\[\s*col\s*\]\s*!=\s*num/);
+```
```

Comment:
I interpret this as:

```python
        for i in self.board[row][col]:
            num != i
        # change code above this line
        for row in range(9):
```

However, what are we supposed to do with:
> check whether the value at the current position in the 2D list is not equal to the provided `num`.

Also, `row` is not defined above the section told where to code.

Reply:
We are checking if the number is already present in the column, then it is not valid for insertion in the board.

Also, the complete statement looks like this: `self.board[row][col] != num for row in range(9)` and the value of `row` is coming from the range specified in the preceding `for` loop